src/components/DateRangePicker.js

```diff
@@ -0,0 +1,353 @@
+const React = require('react');
+const Radium = require('radium');
+const moment = require('moment');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const DatePicker = React.createClass({
+  propTypes: {
+    format: React.PropTypes.string,
+    locale: React.PropTypes.string,
+    minimumDate: React.PropTypes.number,
+    onDateSelect: React.PropTypes.func,
+    placeholderText: React.PropTypes.string,
+    primaryColor: React.PropTypes.string,
+    selectedEndDate: React.PropTypes.number,
+    selectedStartDate: React.PropTypes.number,
+    style: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      format: 'MMM D, YYYY',
+      locale: 'en',
+      onDateSelect () {},
+      placeholderText: 'Select A Date Range',
+      primaryColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      currentDate: this.props.selectedStartDate || moment().unix(),
+      showCalendar: false
+    };
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.selectedStartDate && newProps.selectedStartDate !== this.props.selectedStartDate) {
+      this.setState({
+        currentDate: newProps.selectedStartDate
+      });
+    }
+  },
+
+  _handleDateSelect (date) {
+    if (this.state.selectedStartDate === date) {
+      this.state.selectedStartDate = null;
+      return;
+    }
+    if (this.state.selectedEndDate === date) {
+      this.state.selectedEndDate = null;
+      return;
+    }
```

Comment:
This deselects the date after it's been selected. Is there a cleaner way to do this?